### DISCUSSION TITLE
Discussion Regarding Frigate and DoubleTake Integration

### DISCUSSION THREAD

**Discussion by LordNex** [xannor/ha_reolink_rest#20] [Ideas]
My main question is this just a tool to discover and managed the cameras or does this bring each stream into HA?

My setup uses Frigate on a RPi4 with a Coral TPU doing object discovery and when it sees something in its criteria it sends a clip by MQTT to DoubleTake running on HA. Double take then passes it off to 3 different neural networks I have running on a NVIDIA Jetson modules. I have DoubleTake train faces I want remembered, such as my family, so that I can do camera based automations upon valid detection. 

I've currently running several Amcrest AD410s successfully through the system and I purchased a handful ReoLink 520As. As long as I can get the stream into Frigate, Frigate can pass that into HA and my process system. 

I have one 520a mounted, just got to run a line to it quick and it would be up and running. But I've just put it off because I knew it would be like the rest and take some hacking. But seeing this give me a little bit more hopes that I picked the right hardware.

**Comment by xannor** [comment]
This integrations purposes is similar to the amcrest and other camera integrations. To allow home assistant to utilize and manage reolink devices. 

I plan to recreate a lot of the features I helped write into the fwesterberg reolink integration, so home assistant sees the cameras as first class citizens.

**Comment by LordNex** [comment]
> This integrations purposes is similar to the amcrest and other camera integrations. To allow home assistant to utilize and manage reolink devices. 
> 
> 
> 
> I plan to recreate a lot of the features I helped write into the fwesterberg reolink integration, so home assistant sees the cameras as first class citizens.

Very cool. You might take a look at DoubleTake and what they've been doing. I found a program called PhotoPrism that allows you to name individual faces and it will then show you every picture they've been in your entire library. Eventually I'd like to be any to sync those facial algorithms into the training method DoubleTake uses to train AI detectors like DeepStack and CompreFace. I could see this being very handy as there are only a handful of decent cameras out there that work well with Home Assistant. ReoLink and Amcrest being my mains.

**Comment by kmbdk** [comment]
> I plan to recreate a lot of the features I helped write into the fwesterberg reolink integration, so home assistant sees the cameras as first class citizens.

Quick question, do you plan on adding switches to the integration as the Relink ip cam integration has, for enabeling/disabeling notifications and recording, or will you still need to run both modules to do this?

**Comment by xannor** [comment]
I plan on supporting all the "abilities" and settings of the cameras. Currently I am focusing on the most obvious, streams, snaps, and motion, next is video retrieval.

**Comment by LordNex** [comment]
> I plan on supporting all the "abilities" and settings of the cameras. Currently I am focusing on the most obvious, streams, snaps, and motion, next is video retrieval.

So far it's been working beautifully. Thanks a lot for your hard work!
